runtime/lua/vim/lsp.lua

```diff
@@ -2260,6 +2267,22 @@ function lsp.buf_get_clients(bufnr)
   return result
 end
 
+--- Toggle inlay hints for a buffer
+---
+--- @param bufnr (integer|nil): Buffer handle, or nil for current
+function lsp.buf_inlay_hint_toggle(bufnr)
```

Comment:
Sorry for the back and forth, but could you remove this again from this PR. We can follow up on these things later in dedicated PRs.

I think the APIs will need some more discussions as there's quite a few choices to make. (Do we really want a mix of `buf_inlay_hint_*` and `inlay_hints.<xy>`. Do we mirror diagnostics with `.show()`, `.hide()`, `.enable()`, `.disable()`, or something else. Just to mention a few points)

I'd like to keep the scope of this PR small, unless somebody thinks these things will impact the API of the functions that are being added here (`refresh()`, `clear()` and `on_inlayhint()`)


(And it's a bit tedious to review a PR that keeps growing in scope, so I'd ask others to not treat this PR as a wishlist for things to add in addition, but maybe create an issue where we can track these things. Unless of course the feature would impact the design of what's being added here)

Reply:
My original comment was mostly for lack of documentation.

I agree that a generic/consistent API for such "code overlays" is preferable. (But I do think the "toggle" use case is prevalent enough for it to be part of such -- including for diagnostics! -- so users are not forced to write wrappers around `show()`/`hide()` over and over again...)